search/KMPAlgorithm.cpp

```diff
@@ -0,0 +1,58 @@
+#include <iostream>
```

Comment:
check https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md and document/add tests accordingly